state/statemetrics/state.go

```diff
@@ -27,6 +20,20 @@ type State interface {
 	UserAccess(names.UserTag, names.Tag) (permission.UserAccess, error)
 }
 
+// PooledState is a wrapper for State that includes methods to negotiate with
+// the pool that supplied it.
+type PooledState interface {
+	state.PoolHelper
+	State
+}
+
+// StatePool represents a pool of State objects.
+type StatePool interface {
+	SystemState() State
+	Get(modelUUID string) (PooledState, error)
+	GetModel(modelUUID string) (Model, state.PoolHelper, error)
```

Comment:
This does feel like it should be a PooledModel as well, but that can come later.